ui/src/app/course/components/form-elements/input-duration/input-duration.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, forwardRef, Input } from '@angular/core';
+import { ControlValueAccessor, FormControl, NG_VALUE_ACCESSOR, ValidatorFn, Validators } from '@angular/forms';
+
+export const DurationValidators: ValidatorFn[] = [
+	Validators.min(0),
+	Validators.max(999999),
+	Validators.pattern(/^\d+$/),
+];
+
+@Component({
+	selector: 'app-input-duration',
+	templateUrl: './input-duration.component.html',
+	styleUrls: ['./input-duration.component.scss'],
+	providers: [{
+		provide: NG_VALUE_ACCESSOR,
+		useExisting: forwardRef(() => InputDurationComponent),
+		multi: true,
+	}],
+})
+export class InputDurationComponent implements ControlValueAccessor {
+
+	@Input() public formControl: FormControl = new FormControl();
+	@Input() public fieldName: string = 'duration';
```

Comment:
`fieldName` is useless in case you have `formControl`. All things can be done using `formControl`

Reply:
fieldName this is name for show in error messages